app/Http/Controllers/Group/Workspace/TopicController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Controllers\Group\Workspace;
+
+use App\Http\Controllers\Controller;
+use App\Models\Bookmark;
+use App\Models\Group;
+use App\Models\LegiScan\Session;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+
+class TopicController extends Controller
+{
+    public function show(Group $group, Group $workspace, Group $topic)
```

Comment:
Everything that leads up to this controller method needs to be rethought from scratch. This should scream "I haven't thought through my database structure and Model design" in 10-foot-tall neon letters to you. No Controller method should have contents like this.

Ideal Controller method:

- Accept & validate a request
- Perform simple logic or defer to complex objects
- Return a response

This Controller method:

- Figure out masses of confusing nested SQL to build an entire world from scratch and then serve up a hot mess back to the view to figure out with every Model in the database all at once for pretty much no reason.

Reply:
The main issue you have is three variables with the same class?

Reply:
No, it's the entire contents of the controller method.

Reply:
Fixed!